### ISSUE TITLE
new_zoom_delta_algorithm causes 3DSS crash when zooming out on variable power optic

### ISSUE THREAD

**Issue by York2016** [themrdemonized/xray-monolith#120]
Something in new_zoom_delta_algorithm function (introduced in modded exes build 2024.11.10) seems to cause an 'assertion failure' with 3DSS, from time to time. (or at least 3DSS Gamma, not sure about regular 3DSS or 3DSS EFP) 

Sometimes (it's a bit inconsistent when this happens from my experience), when quickly switching between a 1x and variable power 3DSS scope - whether that be via alt-aim or by switching optics in inventory - causes the variable power optic to 'corrupt' fixing it to 1x and then crash, with said assertion failure error, when you zoom out. Weirdly, zooming in does not do this - only zooming out. There is a workaround by just switching to alt aim and back again, but it is annoying. This problem does not occur with fixed power optics ofc, but they do seemingly still receive the 'forced to 1x' part of the bug however.

From limited testing I've seen that setting new_zoom_delta_algorithm to 0, or disabling it in the modded exes options menu, prevents the problem from occurring.

**Comment by themrdemonized** [comment]
I need a log and mdmp file when it happens

**Comment by York2016** [comment]
[xray_crasher_007.log](https://github.com/user-attachments/files/18282286/xray_crasher_007.log)
[xray_crasher_007NoSWM.log](https://github.com/user-attachments/files/18282390/xray_crasher_007NoSWM.log)


I tried to get an mdmp but apparently it crashes so fast that no mdmp is created at all.
The SWM legs error (1st log) is a red herring, i tested it without that mod and it still crashed with a new assertion error (2nd log).

The issue is most apparent upon loading in after a level transition but has happened randomly while roaming around. During this period, you can also 'force' guns that normally don't have alt aim into alt-aim mode, before the game 'remembers' and disables it again after you switch back to normal aim.

**Comment by themrdemonized** [comment]
I've read the logs and the errors aren't related to either new_zoom, nor 3dss in any way, at least directly. You probably have some sort of script error that leads to "busy hands", that can also lead to other errors, at least thats what the logs told me. I can't really help with this information on hands unfortunately.

**Comment by York2016** [comment]
Indeed, just today, several of us on the Gamma Discord have found that it might actually related to https://www.moddb.com/mods/stalker-anomaly/addons/153-dynamic-viewmodel of all things. I have notified the author of that mod already.


However, Andtheherois has notified me that he does not have that mod but yet can still reproduce the crash from time to time, just much more rarely vs the people who have that dynamic viewmodel mod.
